presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -81,7 +81,11 @@ public synchronized void updateMemoryUsage(long bytesAdded)
             return;
         }
 
-        long currentBufferedBytes = bufferedBytes.addAndGet(bytesAdded);
+        long localBufferedBytes = bufferedBytes.get();
+        long currentBufferedBytes = localBufferedBytes + bytesAdded;
+        checkArgument(currentBufferedBytes >= 0, "bufferedBytes (%s) plus delta (%s) would be negative", localBufferedBytes, bytesAdded);
+        bufferedBytes.set(currentBufferedBytes);
```

Comment:
it's no longer atomic. Use `accumulateAndGet` instead, placing the check inside the method.

```
long currentBufferedBytes = bufferedBytes.accumulateAndGet(bytesAdded, OutputBufferMemoryManager::addNonNegative);
```

Reply:
It doesn’t need to be atomic. This is the only write and it is guarded by a lock. We use atomic for visibility, to allow reads without a lock.

Reply:
Thanks for checking. 
I think it could still be more _readable_ my way, but up to you

Reply:
Agreed. I switched to `updateAndGet()` with a lambda so that the error message was localized to the operation. Much nicer than what I had before. Thanks